### DISCUSSION TITLE
How to filter by id (or based on multiple columns)?

### DISCUSSION THREAD

**Discussion by heberleh** [lineupjs/lineupjs#571] [Q&A]
Use case: a different type of filter/search is implemented that is based on multiple columns. This doesn't fit the idea of filter per column. On LineUp so far I understand that we can filter using Regex or so with a `Column` object.

However, if we have a hidden column (e.g., `_id`), is it possible to filter it natively using some method?

One solution is to create a new instance of LineUp from scratch, which sounds inefficient since all rankings, columns etc need to be rebuilt. This is the way it is implemented in CIME.

Is there a better solution for filtering or modifying data? I.g. keeping columns but changing what are the rows that are part of lineup (with new data or just filtered rows).

**Comment by thinkh** [comment]
Yes, good idea! I think what you are requesting is covered in issue https://github.com/lineupjs/lineupjs/issues/498.

**Reply by heberleh** [reply]
I actually created a column ID and do the search outside, the IDs that match, then I filter by ID.

Actually, I stopped using Lineup filtering because it's way too slow, so now I reload lineup entirely, it's faster because it loads with less data and even reloading the entire data was fine.

I don't know why the regex by IDs is so slow.

I also missed to filter by hidden column, that's another reason to reload Lineup instead of use it's filters.

The con is to lose the setup made by the user and the colors of the categories,etc, which changes if the range of data is different.

It was difficult to find out how to do all of this, but after mining examples, CIME and Lineup's source code I was able to find how to search column, row, and change data. To access some parts I had to @ts-ignore. Documentation needs improvements.

**Comment by sgratzl** [comment]
for global filtering I usually directly set a filter in the data provider:

https://github.com/lineupjs/lineupjs/blob/93e18c26a61797342b4953a2d6b6a9558ce4a177/src/provider/LocalDataProvider.ts#L108-L115

i.e. lineup.data.setFilter(...)
